File: doc-examples/example-python/src/main/python/example/ReactiveProfileService.py
from jakarta.inject import Singleton
from micronaut.objectstorage import ReactiveObjectStorageOperations


# tag::class[]
@Singleton
class ReactiveProfileService:

    def __init__(self, object_storage: ReactiveObjectStorageOperations):
        self.object_storage = object_storage
# end::class[]

File: doc-examples/example-java/src/main/java/example/ReactiveProfileService.java
package example;

import io.micronaut.objectstorage.ReactiveObjectStorageOperations;
import jakarta.inject.Singleton;

//tag::class[]
@Singleton
public class ReactiveProfileService {

    private final ReactiveObjectStorageOperations<?, ?, ?> objectStorage;

    public ReactiveProfileService(ReactiveObjectStorageOperations<?, ?, ?> objectStorage) {
        this.objectStorage = objectStorage;
    }
}
//end::class[]
